### PR TITLE
TCM-452 | Prepare loading screen

### PR DESCRIPTION
**Pull request by devender197**
### Summary

- [x] Loading Screen added
- [x] Styled according to TRX design.

### Issue

[TCM-452](https://hyfnla.atlassian.net/browse/TCM-452)

### Video/Screenshot

https://user-images.githubusercontent.com/24737954/120244287-b2d45780-c21e-11eb-874c-42dcdb0524c0.mov

### CODE REVIEW

File: app/src/main/java/com/trx/consumer/screens/loading/LoadingStateInterface.kt [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package com.trx.consumer.screens.loading
+
+import com.trx.consumer.screens.loading.LoadingViewState
+
+interface LoadingStateInterface {
```
Comment (VivekV95) [inline app/src/main/java/com/trx/consumer/screens/loading/LoadingStateInterface.kt:5]: Is this needed? Not currently being used

**Review by VivekV95 — CHANGES_REQUESTED** [review]
Looking good, just one comment

File: app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -289,4 +290,6 @@
 
     <string name="videos_related_label">Related Workouts</string>
     <string name="videos_btn_primary_label">Start Workout</string>
+
+    <string name="title_loading_fragment">Checking availability …</string>
```
Comment (onodude) [inline app/src/main/res/values/strings.xml:294]: please display “Loading…“ for the state launch state

Comment (devender197) [reply app/src/main/res/values/strings.xml]: [Done](https://github.com/hyfn/trx-consumer-android/pull/91/commits/291b37891867ea91c0fdc5d97eaf23d336d56f58#diff-5e01f7d37a66e4ca03deefc205d8e7008661cdd0284a05aaba1858e6b7bf9103R294)

File: app/src/main/java/com/trx/consumer/screens/loading/LoadingStateInterface.kt (hunk 1 above)
Comment (devender197) [reply app/src/main/java/com/trx/consumer/screens/loading/LoadingStateInterface.kt]: [Removed](https://github.com/hyfn/trx-consumer-android/pull/91/commits/291b37891867ea91c0fdc5d97eaf23d336d56f58#diff-13eb583ed382715c1e02bf6236daa645367c678dff5ea1e4dd81ae2e210bb220L1)

**Comment by devender197** [comment]
@onodude  @VivekV95 Thank you for review. PR suggestion implemented. Please let me know for further enhancements.

**Review by VivekV95 — APPROVED** [review]

**Review by onodude — APPROVED** [review]
thanks!

**Pull request merged**